tools/clang/unittests/HLSLExec/ExecutionTest.cpp

```diff
@@ -4416,7 +4415,7 @@ TEST_F(ExecutionTest, ATOWriteMSAATest) {
 }
 
 // Used to determine how an out of bounds offset should be converted
-#define CLAMPOFFSET(offset) (((unsigned)(offset) << 28) >> 28)
+#define CLAMPOFFSET(offset) (((int)((unsigned)(offset) << 28)) >> 28)
```

Comment:
This makes my brain hurt.  Would a function be clearer here?

Reply:
Changed into function.